custom-war-packager-lib/src/main/java/io/jenkins/tools/warpackager/lib/util/MavenHelper.java

```diff
@@ -143,6 +143,7 @@ public void downloadJAR(File buildDir, DependencyInfo dep, String version, File
                     DependencyInfo dep = new DependencyInfo();
                     dep.groupId = dependencyData[0].trim();
                     dep.artifactId = dependencyData[1].trim();
+                    dep.type = dependencyData[2].trim();
```

Comment:
The `dependency:list` always inform the type, so here  it cannot be null. In `DependencyInfo` I've checked the nullability just in case it could be used in other way.